File: Problem203/Solution.go
package main

type ListNode struct {
	Val  int
	Next *ListNode
}

func removeElements(head *ListNode, val int) *ListNode {
	dummy := &ListNode{Val: -1, Next: nil}
	p := dummy
	for head != nil {
		if head.Val != val {
			p.Next = head
			p = p.Next
		}
		head = head.Next
	}
	//Tail
	if p.Next != nil && p.Next.Val == val {
		p.Next = nil
	}
	return dummy.Next
}

File: Problem203/Solution.java
public class Solution
{
    public ListNode removeElements(ListNode head, int val)
    {
        ListNode dummy = new ListNode(-1);
        ListNode p = dummy;
        while (head != null)
        {
            if (head.val != val)
            {
                p.next = head;
                p = p.next;
            }
            head = head.next;
        }
        // Tail
        if (p.next != null && p.next.val == val)
            p.next = null;
        return dummy.next;
    }
}